src/planner/binder/expression/bind_macro_expression.cpp

```diff
@@ -9,43 +10,165 @@
 
 namespace duckdb {
 
-void ExpressionBinder::ReplaceMacroParametersRecursive(unique_ptr<ParsedExpression> &expr) {
+bool IsLambdaParameter(vector<unordered_set<string>> &lambda_params, ColumnRefExpression &col_ref) {
+	for (const auto &level : lambda_params) {
+		if (level.find(col_ref.GetColumnName()) != level.end()) {
+			return true;
+		}
+	}
+	return false;
+}
+
+void ExpressionBinder::ReplaceMacroParamInLambda(unique_ptr<ParsedExpression> &expr,
+                                                 vector<unordered_set<string>> &lambda_params) {
+
+	// special-handling for LHS lambda parameters
+	// we do not replace them, and we add them to the lambda_params vector
+	auto &lambda_expr = expr->Cast<LambdaExpression>();
+	lambda_params.emplace_back();
+
+	// single parameter, add to lambda_params
+	if (lambda_expr.lhs->expression_class == ExpressionClass::COLUMN_REF) {
+		auto column_ref = lambda_expr.lhs->Cast<ColumnRefExpression>();
+		lambda_params.back().emplace(column_ref.GetColumnName());
```

Comment:
This looks somewhat duplicated from `ExtractParameterList` - can the code from there be re-used?